src/main/java/org/kuali/kfs/gl/batch/service/impl/CuScrubberProcessImpl.java

```diff
@@ -0,0 +1,312 @@
+                }
+                return new TransactionError(costShareEntry, m);
+            }
+
+            createOutputEntry(costShareSourceAccountOffsetEntry, OUTPUT_GLE_FILE_ps);
+            scrubberReport.incrementCostShareEntryGenerated();
+
+            scrubCostShareAmount = KualiDecimal.ZERO;
+        } catch (IOException ioe) {
+            LOG.error("generateCostShareEntries() Stopped: " + ioe.getMessage());
+            throw new RuntimeException("generateCostShareEntries() Stopped: " + ioe.getMessage(), ioe);
+        }
+        LOG.debug("generateCostShareEntries() successful");
+        return null;
+    }
+
+    /**
+     * Helper method for configuring the chart, account and sub-account on the cost-share-source-account entry,
+     * using the cost share account's continuation account or descendant (up to a depth of 10) in the
+     * event of a closed cost share account.
+     * 
+     * @param costShareSourceAccountEntry The origin entry to configure.
+     * @param scrubbedEntryA21SubAccount The A21 sub-account from the original scrubbed origin entry.
+     * @return A TransactionError if a valid cost share or continuation account could not be found, null otherwise.
+     */
+    protected TransactionError setupEntryWithPotentialContinuation(OriginEntryFull costShareSourceAccountEntry, A21SubAccount scrubbedEntryA21SubAccount) {
+        Account costShareAccount = accountingCycleCachingService.getAccount(
+                scrubbedEntryA21SubAccount.getCostShareChartOfAccountCode(), scrubbedEntryA21SubAccount.getCostShareSourceAccountNumber());
+        if (ObjectUtils.isNotNull(costShareAccount) && costShareAccount.isClosed()) {
+            // Cost share source account is closed; check for a valid continuation account.
+            Account continuationAccount = costShareAccount;
+            for (int i = 0; i < CONTINUATION_ACCOUNT_DEPTH_LIMIT && ObjectUtils.isNotNull(continuationAccount) && continuationAccount.isClosed(); i++) {
+                continuationAccount = accountingCycleCachingService.getAccount(
+                        continuationAccount.getContinuationFinChrtOfAcctCd(), continuationAccount.getContinuationAccountNumber());
+            }
+            if (ObjectUtils.isNull(continuationAccount) || costShareAccount == continuationAccount || continuationAccount.isClosed()) {
+                // Could not find a valid Cost Share continuation account; return an error.
+                return new TransactionError(costShareSourceAccountEntry, new Message(MessageFormat.format(
+                        configurationService.getPropertyValueAsString(CUKFSKeyConstants.ERROR_CSACCOUNT_CONTINUATION_ACCOUNT_CLOSED),
+                                scrubbedEntryA21SubAccount.getCostShareChartOfAccountCode(), scrubbedEntryA21SubAccount.getCostShareSourceAccountNumber()),
+                        Message.TYPE_FATAL));

```

Comment:
Is this error misleading? There are two other conditions that could cause it besides the account being closed?